travvy/src/components/SelectFile.vue

```diff
@@ -4,25 +4,74 @@
       <span v-if="value">The File you have Selected is: {{value.name}}</span>
       <span v-else>Select File</span>
     </div>
-    <input type="file" @change="handleFileChange"/>
+    <div>
+    <input type="file" @change="previewImage" accept="image/*">
+    </div>
+    <div>
+      <h2>Progress: {{uploadValue.toFixed()+"%"}}
+        <progress id="progress" :value="uploadValue" max="100"></progress></h2>
+    </div>
+    <div v-if="imageData!=null">
+      <img class="preview" :src="picture">
+      <br>
+      <button class="selectFiles" @click="onUpload"><span>Upload File</span></button>
+    </div>
+
   </label>
 </template>
 
 <script>
+import firebase from 'firebase';
+
 export default {
+  name: 'Upload',
+
+  data() {
+    return{
+      imageData: null,
+      picture: null,
+      uploadValue: 0
+    }
+  },
+
   props: {
     value: File
   },
 
   methods: {
     handleFileChange(e) {
       this.$emit('input', e.target.files[0])
+    },
+
+
+    previewImage(event) {
+      this.uploadValue=0;
+      this.picture=null;
+      this.imageData = event.target.files[0];
+    },
+
+    onUpload() {
+      this.picture=null;
```

Comment:
Next sprint we can insert the user email along with the image so that it can be extracted